Additional LLILC debugging aids

richardlford opened the pull request:
The changes makes the following improvements to the
LLILC debugging facilities:

1. A make_sln.py script is added which can be used to create
a Visual Studio 2015 solution that can be used to debug
a managed application using LLILC. Optionally the
script can invoke Visual Studio on the created solution.
The handles setting up all the required environment
variables for making LLILC the alternate JIT.

2. A --windbg-and-args option was added to the
llilc_run.py script to enable debugging the
application being run using windbg.

3. In order to ensure that empty values for
LLILC configuration variables are equivalent to
unspecified environment variables, the
code that LLILC uses to check some configuration
variables was refactored to require that
the feature is only enabled if the variable
is both non-null and non-empty.

4. The Debugging.md documentation was updated
to describe these features.

5. The llilc_run.py and make_sln.py were modified so they
run successfully on either Python 2.7 or Python 3.4.

richardlford commented:
@adiaaida, PTAL.

adiaaida commented on test/make_sln.py:
```diff
@@ -0,0 +1,294 @@
+        error_level = RunCommand(command)
+
+    if llilcverbose:
+        log ('Exiting llilc_run.py with exit code ', error_level)
```
Is this supposed to be make_sln.py, not llilc_run.py?

richardlford replied:
You are right. I'll fix that.

adiaaida commented:
Other than the one nit, looks fine to me.

richardlford commented:
I squashed and pushed the log message.

adiaaida commented:
We are going to be switching to python 2.7 as the default version of Python for LLILC. Can you make sure this runs appropriately when run with 2.7?

richardlford commented:
I modified make_sln.py as well as llilc_run.py so they run successfully on both Python 2.7 and Python 3.4. I also added an additional documentation not about the debugger type in the solutions.

Since the previous version passed testing
(http://dotnet-ci.cloudapp.net/job/dotnet_llilc_prtest/990/)
and this change only changed python scripts that are not used lab tests, I'm
not going to wait for the lab job to complete before merging.
